deploy/crds/kieapp.crd.yaml

```diff
@@ -10,1768 +10,3302 @@ spec:
+                      image:
+                        description: The image to use for Process Instance Migration
+                        type: string
+                      imageTag:
+                        description: The image tag to use for Process Instance Migration.
+                        type: string
+                    type: object
                   servers:
-                    type: array
                     description: Configuration of the each individual KIE server
-                    minItems: 1
                     items:
-                      type: object
-                      description: KIE Server configuration
+                      description: KieServerSet KIE Server configuration for a single
+                        set, or for multiple sets if deployments is set to >1
                       properties:
-                        name:
-                          type: string
-                          description: Server name
-                        id:
-                          type: string
-                          description: Server ID
-                        image:
-                          type: string
-                          description: The image to use for server.
-                        imageTag:
-                          type: string
-                          description: The image tag to use for server.
-                        deployments:
-                          type: integer
-                          format: int
-                          description: Number of Server sets that will be deployed
-                        replicas:
-                          type: integer
-                          format: int32
-                          description: Replicas to set for the DeploymentConfig
-                        keystoreSecret:
-                          type: string
-                          description: Keystore secret name
-                        from:
-                          type: object
-                          description: Image definition to use for all the servers
-                          required:
-                            - kind
-                            - name
-                          properties:
-                            kind:
-                              type: string
-                              description: Object kind
-                              enum:
-                                - ImageStreamTag
-                                - DockerImage
-                            name:
-                              type: string
-                              description: Object name
-                            namespace:
-                              type: string
-                              description: Namespace where the object is located
                         build:
-                          type: object
-                          description: Configuration of build configs for immutable KIE servers
-                          oneOf:
```

Comment:
I think I've got everything replicated from our old CRD, except for a couple of these `oneOf` schema options. I've yet to find a validation marker we can use to generate this section. @ruromero how critical are these?